### PR TITLE
vdirsyncer: deprecate vdirsyncerStable

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
```
looking for now-outdated files... none found
pickling environment... done
checking consistency... /build/khal-0.10.2/doc/source/man.rst: WARNING: document isn't included in any toctree
done
writing... khal.1 { usage configure standards faq license } /build/khal-0.10.2/doc/source/usage.rst:53: WARNING: unknown option: --color
/build/khal-0.10.2/doc/source/usage.rst:53: WARNING: unknown option: --color
/build/khal-0.10.2/doc/source/usage.rst:395: WARNING: unknown option: --interactive
/build/khal-0.10.2/doc/source/usage.rst:395: WARNING: unknown option: -i
/build/khal-0.10.2/doc/source/configure.rst:17: WARNING: unknown option: -c
path/to/config
done
build succeeded, 7 warnings.

The manual pages are in build/man.

make: /bin/sh: No such file or directory
make: *** [Makefile:133: man] Error 127
make: Leaving directory '/build/khal-0.10.2/doc'
```

**Comment by magnetophon** [comment]
@SuperSandro2000 Thanks for the feedback, but I am not getting those errors.
When I run ``nix-env -f $NIXPKGS -iA khal``, I *do* get ``build succeeded, 7 warnings.``, but I don't get those make errors.

Both ``vdirsyncer sync`` and ``khal list -o`` work as expected here.

How did you install khal?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -24016,7 +24016,7 @@ in
 
   vdirsyncer = with python3Packages; toPythonApplication vdirsyncer;
 
-  vdirsyncerStable = with python3Packages; toPythonApplication vdirsyncerStable;
+  vdirsyncerStable  = throw "vdirsyncerStable is not supported anymore. Use vdirsyncer instead. For more info, see: https://github.com/NixOS/nixpkgs/issues/103026#issuecomment-723428168";
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:24019]: this should be moved into `pkgs/top-level/aliases.nix`

**Comment by jonringer** [comment]
@SuperSandro2000 is that on darwin?

**Review by loewenheim — APPROVED** [review]
Looks good!

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (magnetophon) [reply pkgs/top-level/all-packages.nix]: Done.

**Review by loewenheim — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 is that on darwin?

yes but now it fails with 

```
tests/khalendar_test.py .............................                    [ 58%]                                                                                                   [20/48634]tests/khalendar_utils_test.py .......................................    [ 71%]                                                                                                             tests/parse_datetime_test.py ........................................... [ 86%]                                                                                                             ....                                                                     [ 87%]                                                                                                             tests/settings_test.py ............                                      [ 91%]                                                                                                             tests/terminal_test.py ...                                               [ 92%]                                                                                                             tests/utils_test.py .......                                              [ 95%]                                                                                                             tests/vdir_test.py X..                                                   [ 96%]                                                                                                             tests/vtimezone_test.py ...                                              [ 97%]                                                                                                             tests/ui/test_calendarwidget.py ...                                      [ 98%]                                                                                                             tests/ui/test_editor.py s...                                             [ 99%]                                                                                                             tests/ui/test_widgets.py .                                               [100%]                                                                                                                                                                                                                                                                                                         =================================== FAILURES ===================================                                                                                                            ___________________ test_vertical_month_unicode_weekdeays_gr ___________________                                                                                                                                                                                                                                                                                                            def test_vertical_month_unicode_weekdeays_gr():                                                                                                                                                 try:                                                                                                                                                                                            locale.setlocale(locale.LC_ALL, 'el_GR.UTF-8')                                                                                                                                              vert_str = vertical_month(month=12, year=2011,                                                                                                                                                                        today=dt.date(2011, 12, 12))
            # on some OSes, Greek locale's abbreviated day of the week and
            # month names have accents, on some they haven't
>           assert strip_accents('\n'.join([line.lower() for line in vert_str])) == \
                '\n'.join(example_gr)
E               AssertionError: assert '\x1b[1m    δ...  1  2  3  4 ' == '\x1b[1m     ...  1  2  3  4 '
E                 -      δε τρ τε πε πα σα κυ
E                 ?         -
E                 +     δε τρ τε πε πα σα κυ
E                 - δεκ  28 29 30  1  2  3  4
E                 ?         -
E                 + δεκ 28 29 30  1  2  3  4
E                 -       5  6  7  8  9 10 11 ...
E
E                 ...Full output truncated (39 lines hidden), use '-vv' to show

tests/cal_display_test.py:368: AssertionError
_________________________ test_vertical_month_abbr_fr __________________________

    def test_vertical_month_abbr_fr():
        # see issue #653
        try:
            locale.setlocale(locale.LC_ALL, 'fr_FR.UTF-8')
            vert_str = vertical_month(month=12, year=2011,
                                      today=dt.date(2011, 12, 12))
>           assert '\n'.join(vert_str) == '\n'.join(example_fr)
E           AssertionError: assert '\x1b[1m    L...  1  2  3  4 ' == '\x1b[1m     ...  1  2  3  4 '
E             -       lu ma me je ve sa di
E             ?         ^^^  ^  ^  ^  ^  ^  ^
E             +     Lu Ma Me Je Ve Sa Di
E             ?         ^  ^  ^  ^  ^  ^  ^
E             - déc.  28 29 30  1  2  3  4
E             ?        --
E             + déc 28 29 30  1  2  3  4 ...
E
E             ...Full output truncated (40 lines hidden), use '-vv' to show

tests/cal_display_test.py:391: AssertionError
=========================== short test summary info ============================
FAILED tests/cal_display_test.py::test_vertical_month_unicode_weekdeays_gr - ...
FAILED tests/cal_display_test.py::test_vertical_month_abbr_fr - AssertionErro...
============= 2 failed, 285 passed, 2 skipped, 2 xpassed in 7.02s ==============
```

**Review by DamienCassou — CHANGES_REQUESTED** [review]
Thank you for your work. When I read the changes made by the commit in this PR, I see:

- `stable.nix` was deleted and `vdirsyncerStable` is deprecated;
- `default.nix` is changed significantly.

I think the story your commits should tell is something like:

- what was in `default.nix` is deleted
- `stable.nix` becomes the new default

To tell this story, I think I would decompose the PR into 2 commits:

1. The first commit deletes `default.nix` and makes `vdirsyncer` points to `stable.nix`
2. The second commit renames `stable.nix` into `default.nix`

The deprecation of `vdirsyncerStable` can happen in any of these 2 commits or a third one. I would put it in a third commit because I like small and atomic commits but that's up to you.

**Comment by magnetophon** [comment]
@DamienCassou Like this?

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -643,6 +643,7 @@ mapAliases ({
   v8_3_16_14 = throw "v8_3_16_14 was removed in 2019-11-01: no longer referenced by other packages";
   valadoc = throw "valadoc was deprecated on 2019-10-10: valadoc was merged into vala 0.38";
   vamp = { vampSDK = vamp-plugin-sdk; }; # added 2020-03-26
+  vdirsyncerStable  = throw "vdirsyncerStable is not supported anymore. Use vdirsyncer instead. For more info, see: https://github.com/NixOS/nixpkgs/issues/103026#issuecomment-723428168";
```
Comment (gebner) [inline pkgs/top-level/aliases.nix:646]:
```suggestion
  vdirsyncerStable  = vdirsyncer; # added 2020-11-08, see https://github.com/NixOS/nixpkgs/issues/103026#issuecomment-723428168
```

This really shouldn't throw, that makes it hard to have the same configuration for master and stable.

Comment (magnetophon) [reply pkgs/top-level/aliases.nix]: Done.

**Comment by DamienCassou** [comment]
> Like this?

Exactly, great job. This is so much easier to understand, thank you. I would only add a sentence to the first commit explaining why the file is deleted.

**Comment by magnetophon** [comment]
> I would only add a sentence to the first commit explaining why the file is deleted.

How would you word it?

**Comment by DamienCassou** [comment]
>>I would only add a sentence to the first commit explaining why the file is deleted.
>How would you word it?

```
vdirsyncer: delete default.nix, point vdirsyncer to stable.nix

TLDR: default.nix was pointing to an unmaintained code base whereas stable.nix is up-to-date and maintained.

History
1. At first their was one Python version of vdirsyncer that had been working fine for years. Then, maintenance decreased and the package was marked as broken in nixpkgs.
2. The original author (@untitaker on github.com) of vdirsyncer decided to re-implement (part of) vdirsyncer in Rust. Nixpkgs made `vdirsyncer` point to the Rust version and renamed the Python historical version to `vdirsyncerStable`.
3. Eventually, @untitaker gave up on the Rust version.
4. Someone else (@WhyNotHugo on github.com) decided to take over maintenance of the Python version.
5.  Mario Rodas (@marsam on github) and Damien Cassou updated the `vdirsyncerStable` to point to the work of @WhyNotHugo and mark the package as working again.
```

**Comment by magnetophon** [comment]
@DamienCassou Thanks for the text!

**Review by loewenheim — APPROVED** [review]

File: pkgs/top-level/aliases.nix [hunk 3]
```diff
@@ -643,6 +643,7 @@ mapAliases ({
   v8_3_16_14 = throw "v8_3_16_14 was removed in 2019-11-01: no longer referenced by other packages";
   valadoc = throw "valadoc was deprecated on 2019-10-10: valadoc was merged into vala 0.38";
   vamp = { vampSDK = vamp-plugin-sdk; }; # added 2020-03-26
+  vdirsyncerStable  = vdirsyncer; # added 2020-11-08, see https://github.com/NixOS/nixpkgs/issues/103026#issuecomment-723428168
```
Comment (jonringer) [inline pkgs/top-level/aliases.nix:646]: to have a true deprecation
```suggestion
  vdirsyncerStable  = lib.warn "vdirsyncerStable has been deprecated, please use vdirsyncer" vdirsyncer; # added 2020-11-08, see https://github.com/NixOS/nixpkgs/issues/103026#issuecomment-723428168
```

Comment (magnetophon) [reply pkgs/top-level/aliases.nix]: :+1:

Comment (jonringer) [reply pkgs/top-level/aliases.nix]: sorry, ofborg does not like this, and will fail if something is written to stderr

```
$ nix-env -f /home/jon/.cache/nixpkgs-review/pr-103073/nixpkgs -qaP --xml --out-path --show-trace --meta
trace: warning: vdirsyncerStable has been deprecated, please use vdirsyncer
4 packages updated:
khal python37Packages.vdirsyncer (2018-08-05 → 0.16.8) python38Packages.vdirsyncer (2018-08-05 → 0.16.8) vdirsyncer (2018-08-05 → 0.16.8)

2 packages removed:
python3.7-vdirsyncer (†0.16.8) python3.8-vdirsyncer (†0.16.8)
```

Comment (jonringer) [reply pkgs/top-level/aliases.nix]: what you had originally would probably be best

Comment (magnetophon) [reply pkgs/top-level/aliases.nix]: Heh.
How's this?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103073` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>khal</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python37Packages.vdirsyncer</li>
    <li>vdirsyncer (python38Packages.vdirsyncer)</li>
  </ul>
</details>

File: pkgs/top-level/aliases.nix (hunk 3 above)
Comment (jonringer) [reply pkgs/top-level/aliases.nix]: as @gebner said, it might be best to just alias it for now.

After the 21.03 release, maybe make it into a throw

Comment (magnetophon) [reply pkgs/top-level/aliases.nix]: Hmmm, when you said originally, I thought you meant the first version....
So you mean this, right?
```nix
  vdirsyncerStable  = vdirsyncer; # added 2020-11-08, see https://github.com/NixOS/nixpkgs/issues/103026#issuecomment-723428168
```

Comment (jonringer) [reply pkgs/top-level/aliases.nix]: Yea

Comment (magnetophon) [reply pkgs/top-level/aliases.nix:646]: OK, done.

**Review by loewenheim — APPROVED** [review]

**Comment by peterhoeg** [comment]
All good - tested here!

**Comment by wirew0rm** [comment]
This seems to have broken the hydra build (and it won't build for me locally either)?

First evaluation after the merge broke, subsequent evaluations use the cached failure:
https://hydra.nixos.org/build/130683519#tabs-summary

Looking at the [log](https://hydra.nixos.org/build/130683519/nixlog/1), the error seems to be the same as the one I'm seeing locally and which was also discussed previously in [this nixpkgs PR](https://github.com/NixOS/nixpkgs/pull/91918) and [this upstream Issue](https://github.com/pimutils/vdirsyncer/issues/837).

If I locally disable the flaky test, it works for me locally:
```diff
--- a/pkgs/development/python-modules/vdirsyncer/default.nix
+++ b/pkgs/development/python-modules/vdirsyncer/default.nix
@@ -53,7 +53,10 @@ buildPythonPackage rec {
-  disabledTests = [ "test_verbosity" ];
+  disabledTests = [
+    "test_verbosity" # was always disabled
+    "test_create_collections" # flaky test sometimes exceeds its deadline
+  ];
```

**Comment by jonringer** [comment]
@wirew0rm if would like to make a PR, I'll be happy to review it
